File: LeetCode_Algorithm/435(Non-overlapping Intervals)/435.js
var eraseOverlapIntervals = function(intervals) {
  // sol1 176ms 99% 73MB 98%
  if (!intervals || intervals.length == 0) {return 0} 
  intervals.sort((a,b) => {
    return a[1] - b[1]
  })
  let cnt = 1; let end = intervals[0][1] 
  for (let i = 1; i < intervals.length; i++) {
    if (intervals[i][0] >= end) {
      end = intervals[i][1]
      cnt += 1
    }
  }
  return intervals.length - cnt
};

console.log(eraseOverlapIntervals([[1,2],[2,3],[3,4],[1,3]]))

File: LeetCode_Algorithm/435(Non-overlapping Intervals)/435.py
class Solution:
    def eraseOverlapIntervals(self, intervals: list[list[int]]) -> int:
        # sol1 1121ms 98% 55.3MB 80%
        if not intervals: return 0
        intervals.sort(key=lambda x : x[1])
        end = intervals[0][1]
        cnt = 1
        for i in range(1, len(intervals)):
            if intervals[i][0] >= end:
                end = intervals[i][1]
                cnt += 1
        return len(intervals) - cnt

print(Solution().eraseOverlapIntervals([[1,2],[2,3],[3,4],[1,3]]))
